### ISSUE TITLE
Bug: Flat config doesn't allow uppercase severities

### ISSUE THREAD

**Issue by mdjermanovic** [eslint/eslint#17570]
### Environment

Node version: v16.14.0
npm version: v8.3.1
Local ESLint version: v8.49.0
Global ESLint version: 
Operating System: windows

### What parser are you using?

Default (Espree)

### What did you do?

<details>
<summary>Configuration</summary>

```
// eslint.config.js

module.exports = [{
    rules: {
        "no-undef": "Error"
    }
}];
```
</details>

```js
foo();
```

### What did you expect to happen?

I'd expect uppercase severities to work. 

eslintrc allows severities such as `"Error"` or `"eRRoR"`. 

flat config validation aims to allow them too:

https://github.com/eslint/eslint/blob/bd7a71fd6b7efb0445393304e2d48c5c06d46a45/lib/config/flat-config-schema.js#L181-L189

### What actually happened?

```
Oops! Something went wrong! :(

ESLint: 8.49.0

Configuration for rule "no-undef" is invalid. Expected severity of "off", 0, "warn", 1, "error", or 2.

You passed 'undefined'.

See https://eslint.org/docs/latest/use/configure/rules#using-configuration-files for configuring rules.
```

### Link to Minimal Reproducible Example

https://stackblitz.com/edit/stackblitz-starters-arttsx?file=eslint.config.js

### Participation

- [X] I am willing to submit a pull request for this issue.

### Additional comments

Validation works as intended, but [`normalizeRuleOptions`](https://github.com/eslint/eslint/blob/bd7a71fd6b7efb0445393304e2d48c5c06d46a45/lib/config/flat-config-schema.js#L115), which is used while merging, returns `undefined`, and then the subsequent validation on the next merge fails. That's why the error message at the end says `You passed 'undefined'`.

**Comment by Rec0iL99** [comment]
Can reproduce 👍.

**Comment by fasttime** [comment]
The constructor option `reportUnusedDisableDirectives` also handles string values in a case-sensitive manner, which seems a bit inconsistent with the intent to allow uppercase severities for rules.

```js
const { FlatESLint } = require('eslint/use-at-your-own-risk');

const eslint = new FlatESLint({
    reportUnusedDisableDirectives: 'Error' // throws ESLintInvalidOptionsError
});
```

**Comment by fasttime** [comment]
There is also a `reportUnusedDisableDirectives` severity setting in [flat config](https://github.com/eslint/rfcs/pull/100). Personally, I think that a consistent handling of severity strings (either case-sensitive or case-insensitive everywhere) would be the least surprising behavior.

**Comment by nzakas** [comment]
> eslintrc allows severities such as "Error" or "eRRoR".

I intentionally did not want to support these in flat config. I was trying to be more strict where possible to clean up some of the weirdness in eslintrc. So, this part isn't a bug, but the "undefined" in the error message definitely is.

My preference is to keep the severities in all places as case-sensitive.

**Comment by nzakas** [comment]
@mdjermanovic are you okay with closing this?

**Comment by mdjermanovic** [comment]
I'd like to fix `flat-config-schema` as in some places it treats rule severities as case-insensitive while in other places as case-sensitive, and that results in the confusing "undefined" in the error message.

Looks like the consensus is that rule severities should be case-sensitive. I'm working on this now.
